components/boards/PostList.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useRouter } from 'next/router';
+import * as BS from './Styled';
+import PostCard from './PostCard';
+import { getArticles } from '@/pages/api/api';
+
+export interface Item {
+    id: number;
+    title: string;
+    content: string;
+    image: string;
+    writer: Writer;
+    likeCount: number;
+    updatedAt: string;
+    createdAt: string;
+}
+
+interface Writer {
+    nickname: string;
+    id: number;
+}
+
+export interface ItemList {
+    totalCount: number;
+    list: Item[];
+}
+
+export default function PostList() {
+    const [itemList, setItemList] = useState<ItemList>({ totalCount: 0, list: [] });
+    const [pageSize, setPageSize] = useState(4); 
+    const [order, setOrder] = useState('recent');
```

Comment:
P2:
아래처럼 state의 타입을 명시해주시면 코드를 파악하는것에 도움이 됩니다.

```suggestion
    const [order, setOrder] = useState<'recent' | 'like'>('recent');
```